festim/hydrogen_transport_problem.py

```diff
@@ -480,13 +486,56 @@ def create_dirichletbc_form(self, bc):
 
         return form
 
+    def create_source_values_fenics(self):
+        """For each source in the model, check if the species and volumes are lists,
+        create one if not, check if each element in species and volumes are
+        festim.Species and festim.VolumeSubdomain1D, then create the value fenics
+        attribute.
+        """
+        for source in self.sources:
+            # iterate through species, find species if name of species is given
+            if not isinstance(source.species, list):
+                source.species = [source.species]
```

Comment:
No, the idea is to _not_ modify the attribute .species!

That's why in my suggestion I created another local variable `source_species` so that the attribute is left untouched

Reply:
What about creating new attributes species_festim and volume_festim and appending the F.Species and F.Volumesubdomain objects to them? Because we will need to access them later on when building the formulation

Reply:
I think for now we can do this same thing when building the formulation. Let's try an minimise the number of attributes overall